### PR TITLE
add old emails to volunteer export report

### LINKED ISSUE #4594: New export report for volunteer email changes
**What type(s) of user does this feature affect?**

 - supervisors
 - admins

**Description**
This ticket depends on [TODO make ticket] which allows volunteers to change (and confirm!) their own emails. Export a report of volunteers who have changed their emails
As always, only export volunteers for the supervisor's current CASA
columns: `old email(s), new email`

**Screenshots of current behavior, if any**  
https://casa-qa.herokuapp.com/reports
Screenshot 2023-02-24 at 11 33 02 AM

_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  
[Link to new theme site](https://casa-new-theme.herokuapp.com/languages/new)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by ShamiTomita**
### What github issue is this PR for, if any?
Resolves #4594

### What changed, and why?
Added the volunteer, old_email attribute array to the Volunteer Export Report

[services/volunteers_emails_export_csv_service.rb](https://github.com/rubyforgood/casa/pull/4809/files#diff-d199e303bedd17510980d6d72e503b7a70159f2cb9fccfbdfd6e408c9be3b391)

[spec/services/volunteers_emails_export_csv_service_spec.rb](https://github.com/rubyforgood/casa/pull/4809/files#diff-22e42c1f201e635b9305495c6d39ce2909b564b7abc845d4f7bfb32ad12fd949)

P.S. This is branching off of my earlier PR #4781, I only changed two files. 

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪
Added an additional factory bot to spec/serivces/volunteer_emails_export_serivce_spec.rb for volunteers with old emails.
Included a relevant expectation

### Screenshots please :)

Screenshot 2023-05-09 at 8 49 33 PM

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -107,4 +107,20 @@ def not_authorized
...
+    notice
+  end
+
+  def check_all_casa_admin_email_change_confirmation_notice(user) # for all casa admins
+    notice = "#{user.role} was successfully updated."
+    if user.saved_changes.include?("unconfirmed_email")
+      notice += " Confirmation Email Sent."
+    end
+    notice
+  end
```
Comment (littleforest) [inline app/controllers/application_controller.rb:125]: These methods look exactly the same to me, are they not? Can we get rid of one and maybe make the name of the method more generic?

File: spec/system/casa_admins/edit_spec.rb [hunk 2]
```diff
@@ -6,27 +6,55 @@
...
 
-      expect(page).to have_text "New admin created successfully"
-      expect(admin.email).to eq expected_email
+      expect(page).to have_text "Casa Admin was successfully updated."
+
       expect(admin.display_name).to eq expected_display_name
       expect(admin.phone_number).to eq expected_phone_number
     end
   end
+  context "with valid email data" do
```
Comment (littleforest) [inline spec/system/casa_admins/edit_spec.rb:28]: picky: add blank line above

**Comment by littleforest** [comment]
@ShamiTomita should #4787 be merged first? Seems like some of the code is duplicated.

**Comment by ShamiTomita** [comment]
> @ShamiTomita should #4787 be merged first? Seems like some of the code is duplicated.

Hi! Yes #4787 should be merged first. I jumped the gun a bit before realizing that #4787 still needed to be approved by a CASA maintainer. I branched this PR from that earlier PR.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (ShamiTomita) [reply app/controllers/application_controller.rb:125]: You're totally right, after refactoring both of them they've basically become identical. I can remove the second one. Would something like `check_unconfirmed_email` be more appropriate?

File: spec/system/casa_admins/edit_spec.rb (hunk 2 above)
Comment (ShamiTomita) [reply spec/system/casa_admins/edit_spec.rb:28]: Thank you for catching it. I had gone through an earlier PR trying to add that formatting and this one must've slipped by me.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (littleforest) [reply app/controllers/application_controller.rb:125]: Sure! Can it be `check_unconfirmed_email_notice` ?

Comment (ShamiTomita) [reply app/controllers/application_controller.rb:125]: Absolutely! I can add those changes to #4787. Thank you!

**Pull request closed without merging**